packages/dd-trace/src/appsec/iast/taint-tracking/plugin.js

```diff
@@ -170,6 +181,37 @@ class TaintTrackingPlugin extends SourceIastPlugin {
     this.taintHeaders(req.headers, iastContext)
     this.taintUrl(req, iastContext)
   }
+
+  _taintDatabaseResult (result, dbOrigin, iastContext = getIastContext(storage.getStore()), name) {
+    if (!iastContext) return result
+
+    let rowsToTaint = this.iastConfig?.dbRowsToTaint
+    if (typeof rowsToTaint !== 'number') {
+      rowsToTaint = 1
+    }
+
+    if (rowsToTaint === 0) return
```

Comment:
you could move this block to `onConfigure` method and resolve it only once setting a `this.rowsToTaint` var

```suggestion
    if (this.rowsToTaint === 0) return
```